lib/src/main/java/com/auth0/jwt/algorithms/RSAAlgorithm.java

```diff
@@ -55,8 +62,65 @@
         this(new CryptoHelper(), id, algorithm, keyProvider);
     }
 
+    @Override
+    public void verifyWithX509(DecodedJWT jwt, EncodeType encodeType, String jwksFile, String pemFile) throws Exception {
+        List<byte[]> byteArrayList = decode(jwt, encodeType);
+        byte[] contentBytes = byteArrayList.get(0);
+        byte[] signatureBytes = byteArrayList.get(1);
+        try {
+            PublicKey publicKey;
+            if(jwksFile != null && !jwksFile.isEmpty() && jwksFile.endsWith(".json")) {
+                String kid = jwt.getKeyId();
+                JwkProvider provider = new UrlJwkProvider(new File(jwksFile).toURI().toURL());
+                Jwk jwk = provider.get(kid);
+                String cert = jwk.getCertificateChain().get(0);
+                try (Writer writer = new BufferedWriter(new OutputStreamWriter(
+                        new FileOutputStream("./jwks.cert"), "utf-8"))) {
+                    writer.write("-----BEGIN CERTIFICATE-----");
+                    writer.append("\n" + cert + "\n");
+                    writer.append("-----END CERTIFICATE-----");
+                }
+
+                FileReader file = new FileReader(pemFile);
+                PemReader reader = new PemReader(file);
+                X509EncodedKeySpec caKeySpec = new X509EncodedKeySpec(reader.readPemObject().getContent());
+                KeyFactory kf = KeyFactory.getInstance("RSA");
+                publicKey = kf.generatePublic(caKeySpec);
+            } else {
+                throw new IllegalArgumentException("Not a proper jwks file");
+            }
+            if (publicKey == null) {
+                throw new IllegalStateException("The given Public Key is null.");
+            }
+            boolean valid = crypto.verifySignatureFor(getDescription(), publicKey, contentBytes, signatureBytes);
+            if (!valid) {
+                throw new SignatureVerificationException(this);
+            }
+        } catch (NoSuchAlgorithmException | SignatureException | InvalidKeyException | IllegalStateException e) {
+            throw new SignatureVerificationException(this, e);
+        }
+    }
+
     @Override
     public void verify(DecodedJWT jwt, EncodeType encodeType) throws Exception {
+        List<byte[]> byteArrayList = decode(jwt, encodeType);
```

Comment:
Why is it that we have to run "decode" again on a "decoded jwt"? I think there could be better naming here.